crypto/dh/dh_check.c

```diff
@@ -155,7 +155,7 @@ int DH_check(const DH *dh, int *ret)
     /* Don't do any checks at all with an excessively large modulus */
     if (BN_num_bits(dh->params.p) > OPENSSL_DH_CHECK_MAX_MODULUS_BITS) {
         ERR_raise(ERR_LIB_DH, DH_R_MODULUS_TOO_LARGE);
-        *ret = DH_MODULUS_TOO_LARGE | DH_CHECK_P_NOT_PRIME;
+        *ret = DH_MODULUS_TOO_LARGE;
```

Comment:
This is a safety setting, please keep it.

Reply:
The return value from the function is 0. That means the *ret does not even pretend to be correct. The problem is not with the backwards compatibility with old versions but keeping applications that do some buggy checking of the *ret value which ignore the return value from the function call safe.

Reply:
If we returned 1 here it would be a different thing but no, we don't.

Reply:
... er, are you arguing against setting `*ret` at all?  'cause by extension, you seem to be arguing a pointlessness regarding the specific flags, i.e. that what anyone might actually check is `ret != 0` if they check it at all.

Reply:
Looking back at the PR where this was created, `DH_CHECK_P_NOT_PRIME` was argued as an *optional* addition.  I would have argued against it, had I looked.

Ref: https://github.com/openssl/openssl/pull/21524#issuecomment-1647788084

Reply:
We would have then to agree to disagree and pass it over to OTC I am afraid.

Reply:
Also strictly speaking this is a fairly unrelated change to this PR so if anything it should be in a separate PR.

Reply:
You have a good point that this doesn't strictly belong in this PR.  I'll move it to another.